js/mail.js

```diff
@@ -1196,6 +1187,35 @@ var Mail = {
 $(document).ready(function() {
 	Mail.UI.initializeInterface();
 
+	// Enables infinite scroll ability.
+	setInterval(function() {
+
+		// Element identifies the element we want to trigger our loading on.
+		//var element = $('.mail_message_summary:nth-last-child(1)');
+		var element = $('#load-more-mail-messages');
+		var outerHeight = element.outerHeight();
+		var offset = element.offset();
+		var scrollLength = $('#mail_messages').scrollTop();
+		var screenHeight = $(document).height();
+		var messageListHeight = scrollLength + offset.top + outerHeight;
+
+		if ((scrollLength + screenHeight) >= messageListHeight) {
+			Mail.UI.messageView.loadMessages('addMore');
+		}
+
+	}, 1000);
```

Comment:
Yes - I have to agree - we shall not implement our own scroll kung-fu.

My guts feeling is telling me that this PR is already trying to many things.

Please split this up into multiple PRs - THX